src/table/Exposure.cc

```diff
@@ -186,61 +186,72 @@ void ExposureFitsWriter::_writeRecord(BaseRecord const & r) {
 //----- ExposureFitsReader ---------------------------------------------------------------------------------
 //-----------------------------------------------------------------------------------------------------------
 
-// A custom FitsReader for ExposureTable/Record - this gets registered with name EXPOSURE, so it should get
-// used whenever we read a table with AFW_TYPE set to that value.
-
-namespace {
-
-class ExposureFitsReader : public io::FitsReader {
+// FitsColumnReader that reads a Persistable subclass T (Wcs, Psf, or Calib here) by using an int
+// column to retrieve the object from an InputArchive and attach it to an ExposureRecord via
+// the Setter member function pointer.
+template <typename T, void (ExposureRecord::*Setter)(PTR(T const))>
+class PersistableColumnReader : public io::FitsColumnReader {
 public:
 
-    explicit ExposureFitsReader(Fits * fits, PTR(io::InputArchive) archive, int flags) :
-        io::FitsReader(fits, archive, flags), _archive(archive)
-    {
-        if (!_archive) {
-            int oldHdu = _fits->getHdu();
-            _fits->setHdu(oldHdu + 1);
-            _archive.reset(new io::InputArchive(io::InputArchive::readFits(*_fits)));
-            _fits->setHdu(oldHdu);
+    static void setup(
+        std::string const & name,
+        io::FitsSchemaInputMapper & mapper
+    ) {
+        auto item = mapper.find(name);
+        if (item) {
+            if (mapper.hasArchive()) {
+                std::unique_ptr<io::FitsColumnReader> reader(new PersistableColumnReader(item->column));
+                mapper.customize(std::move(reader));
+            }
+            mapper.erase(item);
         }
     }
 
-protected:
-
-    virtual PTR(BaseTable) _readTable();
+    PersistableColumnReader(int column) : _column(column) {}
 
-    virtual PTR(BaseRecord) _readRecord(PTR(BaseTable) const & table);
+    virtual void readCell(
+        BaseRecord & record,
+        std::size_t row,
+        fits::Fits & fits,
+        PTR(io::InputArchive) const & archive
+    ) const {
+        int id = 0;
+        fits.readTableScalar<int>(row, _column, id);
+        PTR(T) value = archive->get<T>(id);
+        (static_cast<ExposureRecord&>(record).*(Setter))(value);
+    }
 
-    PTR(BaseTable) _inTable;
-    PTR(io::InputArchive) _archive;
-    SchemaMapper _mapper;
+private:
+    bool _noHeavy;
```

Comment:
``_noHeavy`` isn't used, so far as I can see. (It looks like it belongs in ``SourceFootprintReader``.)